server/src/main/java/io/littlehorse/common/model/metadatacommand/subcommand/PutWorkflowEventDefRequestModel.java

```diff
@@ -0,0 +1,73 @@
+package io.littlehorse.common.model.metadatacommand.subcommand;
+
+import com.google.protobuf.Message;
+import io.grpc.Status;
+import io.littlehorse.common.exceptions.LHApiException;
+import io.littlehorse.common.model.ClusterLevelCommand;
+import io.littlehorse.common.model.getable.global.events.WorkflowEventDefModel;
+import io.littlehorse.common.model.getable.objectId.WorkflowEventDefIdModel;
+import io.littlehorse.common.model.metadatacommand.MetadataSubCommand;
+import io.littlehorse.sdk.common.exception.LHSerdeError;
+import io.littlehorse.sdk.common.proto.PutWorkflowEventDefRequest;
+import io.littlehorse.sdk.common.proto.VariableType;
+import io.littlehorse.sdk.common.proto.WorkflowEventDef;
+import io.littlehorse.server.streams.topology.core.ExecutionContext;
+import io.littlehorse.server.streams.topology.core.MetadataCommandExecution;
+import lombok.Getter;
+
+@Getter
+public class PutWorkflowEventDefRequestModel extends MetadataSubCommand<PutWorkflowEventDefRequest>
+        implements ClusterLevelCommand {
+
+    private String name;
+    private VariableType type;
+
+    public PutWorkflowEventDefRequestModel() {
+        // used by LHSerializable
+    }
+
+    public PutWorkflowEventDefRequestModel(String name, VariableType contentType) {
+        this.name = name;
+        this.type = contentType;
+    }
+
+    @Override
+    public PutWorkflowEventDefRequest.Builder toProto() {
+        return PutWorkflowEventDefRequest.newBuilder().setName(name).setType(type);
+    }
+
+    @Override
+    public void initFrom(Message proto, ExecutionContext context) throws LHSerdeError {
+        PutWorkflowEventDefRequest p = (PutWorkflowEventDefRequest) proto;
+        name = p.getName();
+        this.type = p.getType();
+    }
+
+    @Override
+    public Class<PutWorkflowEventDefRequest> getProtoBaseClass() {
+        return PutWorkflowEventDefRequest.class;
+    }
+
+    @Override
+    public boolean hasResponse() {
+        return true;
+    }
+
+    @Override
+    public WorkflowEventDef process(MetadataCommandExecution executionContext) {
+        WorkflowEventDefIdModel id = new WorkflowEventDefIdModel(name);
+
+        WorkflowEventDefModel old = executionContext.metadataManager().get(id);
+        if (old != null) {
+            if (old.getType() != type) {
+                throw new LHApiException(
+                        Status.ALREADY_EXISTS, "WorkflowEventDef with name %s already exists!".formatted(name));
+            }
+            return old.toProto().build();
```

Comment:
Hm.. this seems incorrect to me. I don't like the fact that we wont let the client know that the task already exist whit the same type.